pkgs/development/libraries/caliper/default.nix

```
@@ -0,0 +1,96 @@
+{ lib, stdenv, fetchurl, python3, cmake, adiak, papi
+, withCuda ? false, cudatoolkit
+, withRocm ? false, rocm-runtime
+, withFortran? false, gfortran
+, withOpenmpi? false, openmpi
+, withMpich ? false, mpich
+, libunwindSupport ? true, libunwind
+, libpfmSupport ? true, libpfm
+, elfutilsSupport ? true, elfutils
+, shared ? !stdenv.hostPlatform.isStatic
+}:
+
+# https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/caliper/package.py
+
+let
+   onOffBool = b: if b then "ON" else "OFF";
+   withMPI = (withMpich == true || withOpenmpi == true);
+in
+
+# We can't have both mpis being used (is this the right way to do this?)
+assert (withMpich && !withOpenmpi) || (!withMpich && withOpenmpi) || (!withMpich && !withOpenmpi);
```

```suggestion
assert (withMpich && !withOpenmpi) || (!withMpich && withOpenmpi) || (!withMpich && !withOpenmpi);
```

Please convert that into a throw with a message which only triggers if both are true.

---

I didn't see how the suggested change above was different?